packages/flutter/lib/src/painting/text_painter.dart

```diff
@@ -343,12 +348,33 @@ class TextPainter {
     }
   }
 
+  bool _lastLayoutWasToMaxIntrinsicWidth = false;
+
   /// Computes the visual position of the glyphs for painting the text.
   void layout() {
     if (!_needsLayout)
       return;
     _paragraph.layout();
     _needsLayout = false;
+    _lastLayoutWasToMaxIntrinsicWidth = false;
+  }
+
+  /// Computes the visual position of the glyphs using the unconstrainted max intrinsic width.
+  void layoutToMaxIntrinsicWidth() {
```

Comment:
Why do we need this at all? Can't we just lay out without a maxWidth constraint, and then use the maxIntrinsicWidth to position it?